File: study/src/main/scala/com/xuxue/spider/thread/Test2.scala
package com.xuxue.spider.thread

import java.util.concurrent.Executors

import scala.concurrent.{Await, ExecutionContext, Future}

/**
  * Created by HanHan on 2016/6/16.
  */
object Test2 extends App{
  import scala.concurrent.duration._
  implicit val ex=ExecutionContext.
    fromExecutor(Executors.newWorkStealingPool(1))

  def calc(name : String,max:Int=0)=(1 to max)
    .map(_*2)
    .map{
      number=>;
        Thread.sleep((1 second).toMillis)
        val cala=number+2
        println(name+ ":"+cala)
        cala
    }

  val f1=Future{
    calc("fut1",10)
  }

  val f2=Future{
    calc("fut2",10)
  }

  Await.ready(Future.sequence(List(f1,f2)),5 minutes)

}

File: study/src/main/java/com/xuxue/spider/test/akka/Test2.java
package com.xuxue.spider.test.akka;

import org.apache.http.client.methods.HttpGet;
import org.apache.http.client.methods.HttpUriRequest;

import java.util.ArrayList;

/**
 * Created by HanHan on 2016/7/8.
 */
public class Test2<T extends HttpUriRequest> {

    ArrayList<T> t=new ArrayList<T>();

    public void add(T tt){
        t.add(tt);
    }

    public static void mian(String[] args){
        Test2 t=new Test2<HttpUriRequest>();
        t.add(new HttpGet());
    }

}
